### PR TITLE
WIP: Windows fixes

### PR DESCRIPTION
**Pull request by gregmagolan**
DO NOT MERGE!

Replaces PR #150

/cc @alexeagle

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by gregmagolan** [comment]
DO NOT MERGE!

/cc @alexeagle

**Comment by gregmagolan** [comment]
This gets `bazel build ...` and `bazel test ...` working on Windows.

Go targets need to be commented out as they break the build. Commenting everything using @io_bazel_rules_go and ts_devserver() is sufficient.

**Comment by gregmagolan** [comment]
Commits mostly speak for themselves.

First commit is a work-around for the vanilla tsc compiler (tsc_library) outputting duplicate .d.ts and .js files which breaks the windows `tsc_wrapped` build. Work-around is to put tsc_wrapped and tsetse into a single `tsc_library` target.

Second work-around is due to tsc requesting libraries at path `C:/lib.xxx.d.ts`. Wasn't able to track down why this path was being requested but the work-around remaps the path to where the lib actually is under node_modules.

**Comment by gregmagolan** [comment]
There are number of path fixes under `internal/tsc_wrapped` to handle `/` vs `\`.

There is a fix for ts_web_test() to use the MANIFEST file under Windows.

There are fixes to the tests under `internal/tsc_wrapped`.

And lastly there are fixes for the tests under `examples` to use the MANIFEST file under Windows.

File: WORKSPACE [hunk 1]
```diff
@@ -16,8 +16,8 @@ workspace(name = "build_bazel_rules_typescript")
 
 git_repository(
     name = "build_bazel_rules_nodejs",
-    remote = "https://github.com/bazelbuild/rules_nodejs",
-    commit = "55c8958ad4d3bdf5dd3320a915e8104edbe3b205",
+    remote = "https://github.com/gregmagolan/rules_nodejs",
+    commit = "0e47e3f485809903924c1b896066230fec1b3ffa",
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply WORKSPACE]: Already done

File: internal/karma/karma.conf.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // Karma configuration
 // GENERATED BY Bazel
+process.env.CHROME_BIN = require('puppeteer').executablePath();
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply internal/karma/karma.conf.js]: ASAIK, Karma looks for CHROME_BIN when it seems Chrome or ChromeHeadless as the browser. This just points Karma to the puppeteer executable in node_modules.

File: examples/es5_output/es5_output_test.sh [hunk 3]
```diff
@@ -1,40 +1,69 @@
 #!/bin/bash
 set -e
 
+MANIFEST="$TEST_SRCDIR/MANIFEST"
+if [ -e "$MANIFEST" ]; then
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply examples/es5_output/es5_output_test.sh]: Yes. There is a MANIFEST on OSX anyway. Haven't checked linux. Oddly, when I was debugging this it wasn't finding the OSX MANIFEST. I'll look into it a bit more.

[the remark this answers is not in the document]
Comment (gregmagolan) [reply examples/es5_output/es5_output_test.sh]: It looks like the MANIFEST file is in the runfiles folder on OSX but it doesn't end up in the runfiles sandbox so it isn't found and loaded.

**Comment by gregmagolan** [comment]
For green build and tests, requires commenting out all the GO targets on Windows and the Bazel 0.11.0 or later (runfiles fix needed for for karma on Windows included in 0.11.0).

**Comment by gregmagolan** [comment]
Replaced by #158

**Pull request closed without merging**